src/library_gl.js

```diff
@@ -3191,6 +3214,9 @@ var LibraryGL = {
       }
     } else {
       view = {{{ makeHEAPView('F32', 'value', 'value+count*4') }}};
+#if WORKAROUND_OLD_WEBGL_UNIFORM_UPLOAD_IGNORED_OFFSET_BUG
```

Comment:
is this needed for every single F32 heap view?

How about creating a helper function to avoid all the code duplication, `makeGLHEAPView` (adding "GL" in the name, and it adds this check).

Reply:
No, it is needed only specifically for these views that it was added to. I would not want to craft any abstractions for this. If I added a `makeGLHEAPView()` (which lives in core `src/parseTools.js`) that we'd have to maintain just for this case, then someone would wonder why only parts of `src/library_gl.js` used `makeGLHEAPView()` and others used regular `makeHEAPView()` and it might leak to being used everywhere. I'd prefer to do this as KISS as possible, not building any complexity on top of this.